peps/pep-0705.rst

```diff
@@ -77,106 +82,310 @@ The problem disappears if we don't have mutator methods in ``Movie``. This could
+
+    from typing import ReadOnly
+
+    class BandAndAlbum(TypedDict):
+        band: str
+        album: ReadOnly[Album]
+
+The ``readonly`` flag is equivalent to marking all entries as ``ReadOnly[]``, guaranteeing no entries are missed by mistake. To avoid potential confusion, it is an error to use both ``readonly=True`` and ``ReadOnly[]``::
+
+    class Band(TypedDict, readonly=True):
+        name: ReadOnly[str]  # Runtime error: redundant ReadOnly qualifier
+        members: Collection[str]
+
+Alternative functional syntax
+-----------------------------
+
+The :pep:`alternative functional syntax <589#alternative-syntax>` for TypedDict also supports these features::
+
+   EntityBand = TypedDict('EntityBand', {'band': Band}, readonly=True, other_keys=False)
+   BandAndAlbum = TypedDict(`BandAndAlbum', {'band': str, 'album': ReadOnly[Album]})
+
+Interaction with other special types
+------------------------------------
+
+``ReadOnly[]`` can be used with ``Required[]``, ``NotRequired[]`` and ``Annotated[]``, in any nesting order:
+
+::
+
+    class Movie(TypedDict):
+        title: ReadOnly[Required[str]]  # OK
+        year: ReadOnly[NotRequired[Annotated[int, ValueRange(-9999, 9999)]]]  # OK
+
+::
 
-    movie: MovieRecord = { "name": "Blade Runner",
-                           "year": 1982 }
-     
-    movie["year"] = 1985  # Fine; mutator methods added in definition
-    movie["name"] = "Terminator"  # Type check error; "name" mutator not declared
+    class Movie(TypedDict):
+        title: Required[ReadOnly[str]]  # OK
+        year: Annotated[NotRequired[ReadOnly[int]], ValueRange(-9999, 9999)]  # OK
+
+This is consistent with the behavior introduced in :pep:`655`.
+
+Inheritance
+-----------
 
-Inheriting, directly or indirectly, from both ``TypedDict`` and ``Protocol`` will continue to fail at runtime, and should continue to be rejected by type checkers.
+To avoid potential confusion, it is an error to have a read-only type extend a non-read-only type::
 
+    class BandAlbumAndLabel(BandAndAlbum, readonly=True):  # Runtime error
+        label: str
 
-Multiple inheritance and Protocol
----------------------------------
 
-* A type that inherits from a ``TypedMapping`` protocol and from a ``Protocol`` protocol must satisfy the protocols defined by both, but is not itself a protocol unless it inherits directly from ``TypedMapping`` or ``Protocol``.
-* A type that inherits from a ``TypedMapping`` protocol and from ``Protocol`` itself is configured as a ``Protocol``. Methods and properties may be defined; keys may not::
+It is also an error to have a type without ``other_keys`` specified, or with it specified as ``True``, extend a type with ``other_keys=False``::
 
-    class A(Movie, Protocol):
-        # Declare a mutable property called 'year'
-        # This does not affect the dictionary key 'year'
-        year: str
+    class Person(NamedDict):  # Runtime error
```

Comment:
It's been a while since the NamedDict class was introduced. Maybe better to keep the examples self-contained within a section so they are easier to understand without having to refer back to a previous section.